recover when lock is stale

anarcat opened the pull request:
Whenever selfspy would crash (or even exit), it would leave behind the
lockfile, which is a problem in itself, which this doesn't fix. But at
least, we should be able to recover when a lockfile is left behind but
the actual PID file is missing.

anarcat commented:
related: #130 and #146.

blueyed commented on selfspy/__init__.py:
```diff
@@ -97,10 +97,14 @@ def check_with_encrypter(password):
+            sys.exit(1)
+        else:
+            print '%s is locked, but no PID file exists. breaking lock' % lockname
+            cfg.LOCK.break_lock()
```
Would be nice to use `print` as a function here, to make the transition to py3 easier.

anarcat replied:
done, although there are tons more places where print is still a statement.

blueyed commented on selfspy/__init__.py:
```diff
@@ -97,10 +97,14 @@ def check_with_encrypter(password):
+            print('Shutting down.')
+            sys.exit(1)
+        else:
+            print('%s is locked, but no PID file exists. breaking lock' % lockname)
```
nitpick: s/breaking lock/Breaking lock./

anarcat replied:
fixed.

anarcat committed:
recover when lock is stale

Whenever selfspy would crash (or even exit), it would leave behind the
lockfile, which is a problem in itself, which this doesn't fix. But at
least, we should be able to recover when a lockfile is left behind but
the actual PID file is missing.
